source/frontend/src/utils/mapPropertyUtils.ts

```diff
@@ -144,3 +147,59 @@ function toMapProperty(
     name: feature?.properties?.NAME,
   };
 }
+
+export function featuresetToMapProperty(
+  featureSet: LocationFeatureDataset,
+  address: string = 'unknown',
+): IMapProperty {
+  const pimsFeature = featureSet.pimsFeature;
+  const parcelFeature = featureSet.parcelFeature;
+  const regionFeature = featureSet.regionFeature;
+  const districtFeature = featureSet.districtFeature;
+
+  const propertyId = pimsFeature?.properties.PROPERTY_ID;
+  const pid = pidFromFeatureSet(featureSet);
+  const pin = pinFromFeatureSet(featureSet);
+  return {
+    propertyId: propertyId ? Number.parseInt(propertyId?.toString()) : undefined,
+    pid: pid ?? undefined,
+    pin: pin ?? undefined,
+    latitude: featureSet.location.lat,
+    longitude: featureSet.location.lng,
+    planNumber: parcelFeature?.properties.PLAN_NUMBER?.toString() ?? undefined,
+    address: address,
+    legalDescription: parcelFeature?.properties?.LEGAL_DESCRIPTION ?? undefined,
+    region: isNumber(regionFeature?.properties.REGION_NUMBER)
+      ? regionFeature?.properties.REGION_NUMBER
+      : RegionCodes.Unknown,
+    regionName: regionFeature?.properties.REGION_NAME ?? 'Cannot determine',
+    district: isNumber(districtFeature?.properties.DISTRICT_NUMBER)
+      ? districtFeature?.properties.DISTRICT_NUMBER
+      : DistrictCodes.Unknown,
+    districtName: districtFeature?.properties.DISTRICT_NAME ?? 'Cannot determine',
+    name: pimsFeature?.properties?.NAME ?? undefined,
+  };
+}
+
+export function pidFromFeatureSet(featureset: LocationFeatureDataset): string | null {
+  if (featureset.pimsFeature !== null) {
```

Comment:
can featureset ever be null or undefined?

Reply:
Not at this point.